fetch.bs

```diff
@@ -5075,6 +5083,34 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
 </ol>
 
 
+<h3 id=tao-check>TAO check</h3>
+
+<p>To perform a <dfn id=concept-tao-check>TAO check</dfn> for a <var>request</var> and
+<var>response</var>, run these steps:
+
+<ol>
+ <li><p>If <var>response</var>'s <a for=request>timing allow failed flag</a> is set, then return
+ failure.
+
+ <li><p>If <var>request</var>'s <a for=request>tainted origin flag</a> is unset and
+ <var>response</var>'s <a for=response>location URL</a>'s <a for=url>origin</a> is
+ <a>same origin</a> with <var>request</var>'s <a for=request>origin</a>, then return success.
```

Comment:
Instead of this, I think we should skip this check when request's response tainting is "basic". Otherwise, when you have A -> B -> A, we won't do a TAO check for the final A, right? That would be problematic.

Reply:
It's possible, if the header says `*` (which arguably is also problematic for credentialed requests, but so be it) or `null` (which is what the `Origin` request header will contain).

Reply:
Hmm ok, I'll ask around to see if someone can explain to me the tainted origin flag within CORS. I'd like to better understand what it solves so that I can see how it could also solve problems with TAO. In the meantime, I'm ok including the 'serializing a request origin' since hopefully this would just be an edge case in the wild, thus not breaking a lot of performance monitoring.

Reply:
@yutakahirano explained reasoning for the tainted origin flag in CORS. While it's hard to see how to apply it to TAO, I'm not opposed to aligning here.

Reply:
Okay, so this still needs changes, right?

Reply:
I've added the check so that the same-origin check is only used when response tainting is "basic". I'm also using the 'serializing a request origin' to obtain the origin, so that the tainted origin flag affects the outcome. What's missing?

Reply:
Well, you're checking location URL and that really only works if the response is a redirect, which isn't checked in this step and I also don't think matters. If from origin A you fetched a redirect from origin A it does not matter whether that redirect goes to A or B. You don't need a TAO check for it, right? Or are we requiring TAO for all redirect responses, even same-origin? (That might make some sense too, given how we don't normally expose redirects.)

Reply:
Ah ok, I think I see what you mean. If I understand correctly, up to this point it suffices to check the response being basic.